modules/common/tls/tls.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2023 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tls
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+)
+
+// TLSService contains server-specific TLS settings
+type TLSService struct {
+	// Server-specific settings
+	SecretName             string `json:"secretName,omitempty"`
+	DisableNonTLSListeners bool   `json:"disableNonTLSListeners,omitempty"`
+}
+
+// TLSCa contains CA-specific settings, which could be used both by services (to define their own CA certificates)
+// and by clients (to verify the server's certificate).
+type TLSCa struct {
+	// CA-specific settings
+	CaSecretName string `json:"caSecretName,omitempty"`
+}
+
+// TLS is ageneric type, which encapsulates both the service and CA configurations
+type TLS struct {
+	Service TLSService `json:"service"`
+	Ca      TLSCa      `json:"ca"`
+}
+
+// NewTLS - initialize and return a TLS struct.
+func NewTLS(clientset *kubernetes.Clientset, namespace string, service TLSService, ca TLSCa) (*TLS, error) {
+	if service.SecretName == "" {
+		service.SecretName = "default-tls-secret"
+	} else {
+		err := EnsureOrCreateSecret(clientset, namespace, service.SecretName, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if ca.CaSecretName == "" {
+		ca.CaSecretName = "default-ca-secret"
+	} else {
+		err := EnsureOrCreateSecret(clientset, namespace, ca.CaSecretName, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &TLS{
+		Service: service,
+		Ca:      ca,
+	}, nil
+}
+
+// EnsureOrCreateSecret - checks if a secret with the given name exists in the specified namespace
+func EnsureOrCreateSecret(clientset *kubernetes.Clientset, namespace, secretName string, data map[string][]byte) error {
```

Comment:
lib-common common module has funcs to manage secret. don't think we need an extra one? also I am not sure if we want to create secrets. we just return an error if the referenced secret with the name does not exist.